src/pudl/metadata/fields.py

```diff
@@ -54,6 +56,84 @@
 See https://www.eia.gov/electricity/data/eia411/#tabs_NERC-3.
 """
 
+CUSTOMER_CLASSES: List[str] = [
```

Comment:
I wasn't 100% clear on the meaning of your original comment on #1214 about how in some cases we'd be able to go through the class interface, and in others we'd need to use the constants directly.

I think it might help to have a little more context in the module docstrings for stuff under `pudl.metadata` to clarify what kinds of things are supposed to go in which modules.

When a constant is only ever used in a single module I've been trying to move the constant into that module (and out of `pudl.constants` -- like the very specific string cleaning dictionaries in `pudl.transform.ferc1` Does that seem like the right approach?

Reply:
As far as using `field.constraints.enum` directly... is the idea that we'd be creating lots of Field and Resource objects all over the place on the fly, in order to access the metadata from other parts of the codebase, just based on the table or column names?

Reply:
How about this: When a constant is only used in one module, it can be defined near the top of that module using UPPERCASE_SNAKE_CASE. Otherwise, the constant should be placed somewhere in `pudl.metadata.*`. Which module exactly I'm not sure, but the same logic could apply: If it is needed in `metadata.field` / `metadata.resources` / `metadata.classes` and nowhere else, it can go there, otherwise it should go e.g. in `metadata.constants` or one of the more specific modules, like `metadata.labels` or `metadata.enums` (which I just added).

Whether and how the transform step should use the Field and Resource classes directly gets into a much larger conversation which should probably be its own epic. It really depends on how you approach the transform step. But the sooner the extracted data matches (by name, if not by content) the metadata, the more you can imagine doing with methods like Field.transform_ds(ds), Resource.tranform_df(df) that draw on built-in enum constraints, value maps (which don't exist but could be added), etc.